contracts/modules/cover/Cover.sol

```diff
@@ -870,15 +661,16 @@ contract Cover is ICover, MasterAwareV2, IStakingPoolBeacon, ReentrancyGuard, Mu
     _capacityReductionRatios = new uint[](productIds.length);
     _initialPrices = new uint[](productIds.length);
 
+    ICoverProducts _coverProducts = coverProducts();
     for (uint i = 0; i < productIds.length; i++) {
       uint productId = productIds[i];
 
-      if (productId >= _products.length) {
+      if (productId >= _coverProducts.productsCount()) {
         revert ProductDoesntExist();
       }
 
-      _initialPrices[i] = uint(_products[productId].initialPriceRatio);
-      _capacityReductionRatios[i] = uint(_products[productId].capacityReductionRatio);
+      _initialPrices[i] = uint(_coverProducts.products(productId).initialPriceRatio);
+      _capacityReductionRatios[i] = uint(_coverProducts.products(productId).capacityReductionRatio);
```

Comment:
There are multiple calls to `coverProducts` here and almost nothing from the current contract. We need to move over this function as well. Additionally the product count should be cached into a local variable outside the for loop.